docs/dg/learningBasics.md

```diff
@@ -0,0 +1,140 @@
+1. In `Edit Confiurations` of **Intellij**, set the current task to be the gradle task `RepoSense [run]`, and click the debugging button on the right to start debugging (Make sure that you import the project as a gradle project according to [the tutorial](https://se-education.org/guides/tutorials/intellijImportGradleProject.html)).
+1. Step into each break point when you think it is necessary to know more about the implementation, and step over otherwise to continue the execution.
+
+<box type="info" seamless>
+
+When tracing through the program execution, you can cross reference the architecture diagram and `javadoc` of the class and method to check your understanding of the procedure.
+</box>
+
+{{ step(3) }} Get started with contributing
+
+Here is a small [task](https://github.com/reposense/RepoSense/issues/1118) for you to get started on contributing to RepoSense back-end.
+
+* The basic idea is to change or remove the validation for the author name tag located in `AnnotatorAnalyzer.java` by modifying the`REGEX_AUTHOR_NAME_FORMAT` argument.
+* You may need to check that the unit and system testing still passes after making the change by running the command `gradlew systemtest` and `gradlew test`.
+
+Here is more explanation about the usage of author tag.
+
+{{ embed("Appendix: **Using `@@author` tags**", "../ug/usingAuthorTags.md") }}
+
+{{ step(4) }} Get familiar with the workflow
+
+1. The next step is to verify that your changes passes the testing and then submit the pull request. You can refer to [the _Workflow_ page](workflow.html).
+
+<!-- ==================================================================================================== -->
+
+## front-end
+
+<box type="info" seamless>
+
+This section is for developers who want to contribute to the back-end of RepoSense. You may skip this section if you want to contribute as a pure front-end developer.
+</box>
+
+The front-end implementation of RepoSense is located in `frontend/src`.
+
+{{ step(1) }} Learn the necessary tools
+
+It is necessary for you to learn the basics of `Vue.js`, `pug`, and `scss` before working on the project. 
+
+  {{ embed('Appendix: tools to learn for front-end development', 'tools.md', level=2) }}
+
+{{ step(3) }} Learn the RepoSense front-end architecture
+
+* You may want to refer to the [front-end architecture](report.html#report-architecture) to understand the relationship
+* Another way for you to understand the front is to use **Vue.js devtools** to inspect the how the various Vue Components interact with each other. You can refer to the [front-end debugging guide](workflow.html#debugging-front-end) for more information.
+
+{{ step(4) }} Get started with contributing
+
+Here is a small [task](https://github.com/reposense/RepoSense/issues/1268) for you to get started on contributing to RepoSense front-end.
```

Comment:
Same as for backend, we should use the issue tags to filter, so that we don't have to keep changing this link.